pkg/reconciler/taskrun/resources/taskspec.go

```diff
@@ -29,9 +29,11 @@ import (
 // ResolvedTask contains the data that is needed to execute
 // the TaskRun.
 type ResolvedTask struct {
-	TaskName string
-	Kind     v1beta1.TaskKind
-	TaskSpec *v1beta1.TaskSpec
+	TaskName  string
+	Kind      v1beta1.TaskKind
+	TaskSpec  *v1beta1.TaskSpec
+	Task      *v1beta1.Task
```

Comment:
Some of this data seems duplicated - e.g. wouldn't TaskSpec/Name be included in Task?

Reply:
In cases like inline task&taskspec already in status, there is no "task" but we have the "taskSpec", so I think we cannot use task here to replace the other fields? Another reason is try to reduce the code changes

Reply:
This is a bit confusing though because there are similar pieces of data and it's not clear when to use one over the other. e.g.

- Task.spec vs  TaskSpec
- Task.ObjectMeta.Name vs TaskName
- Task.TypeMeta.Kind vs Kind

If these are the same, then we should ideally just use a single type (probably Task). If they're different then we should document the differences and explain when you'd want to use each.

Reply:
Emmm, I agree there are similar pieces of data, it looks like some data may be contained in other data and this may get people confused. 
But we probably cannot use `Task.***` to replace 'TaskSpec', `TaskName` and `Kind` in my opinion (correct me if I'm wrong), take this code for example:

https://github.com/tektoncd/pipeline/blob/3d6ec63877258b5d9df2eb31aff30a75388c6536/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L678-L704

the TaskSpec is from *taskRun.Status.TaskSpec, 
the taskName is from pipelineTask.TaskRef.Name and kind is from pipelineTask.TaskRef.Kind. 

I will add doc strings for these fields to clarify